From horms@verge.net.au  Thu Jan 14 16:14:41 2010
From: Simon Horman <horms@verge.net.au>
Date: Wed, 23 Dec 2009 19:54:50 +1100
Subject: Staging: rt2870: rtusb_probe() should be in section __devinit
To: Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <20091223085552.225938423@vergenet.net>


$ make CONFIG_DEBUG_SECTION_MISMATCH=y
WARNING: drivers/staging/rt2870/rt2870sta.o(.text+0x2f4c5): Section
mismatch in reference from the function rtusb_probe() to the function .devinit.text:rt2870_probe()
The function rtusb_probe() references
the function __devinit rt2870_probe().
This is often because rtusb_probe lacks a __devinit
annotation or the annotation of rt2870_probe is wrong.

Signed-off-by: Simon Horman <horms@verge.net.au>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/usb_main_dev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rt2860/usb_main_dev.c
+++ b/drivers/staging/rt2860/usb_main_dev.c
@@ -296,7 +296,7 @@ static BOOLEAN USBDevConfigInit(IN struc
 
 }
 
-static int rtusb_probe(struct usb_interface *intf,
+static int __devinit rtusb_probe(struct usb_interface *intf,
 		       const struct usb_device_id *id)
 {
 	struct rt_rtmp_adapter *pAd;
